src/main/java/com/example/purerestapi/service/UserService.java

```diff
@@ -0,0 +1,48 @@
+package com.example.purerestapi.service;
+
+import com.example.purerestapi.exception.UserNotFoundException;
+import com.example.purerestapi.entity.Note;
+import com.example.purerestapi.entity.User;
+import com.example.purerestapi.repository.UserRepository;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+
+  private final UserRepository userRepository;
+  public void addUser(User user) {
+    userRepository.addUser(user);
+  }
+
+  public User getUser(String id) {
+    return userRepository.getUser(id).orElseThrow(UserNotFoundException::new);
+  }
+
+  public void deleteUser(String id) {
+    userRepository.deleteUser(id);
+  }
+
+  public List<User> getAll() {
+    return userRepository.getAll();
+  }
+
+  public User addNote(String userUuid, Note note) {
+    var user = this.getUser(userUuid);
+    if (user.getNotes() == null) {
+      user.setNotes(new ArrayList<>());
+    }
+    note.setUserUuid(userUuid);
+    user.getNotes().add(note);
```

Comment:
You can encapsulate this logic in the  User class via the helper method and use it here instead(e.g. `user.addNote(note)`)

Reply:
thanks for suggestion, valid point fixed)